### PR TITLE
WIP: Refactoring some of the functions in roi.py

### PR DESCRIPTION
**Pull request by ericdill**
Companion PR to Nikea/xray-vision#68

### CODE REVIEW

**Comment by sameera2004** [comment]
Looks like this is the first time we are sing pandas here

File: skxray/core/roi.py [hunk 1]
```diff
@@ -357,39 +356,35 @@ def roi_pixel_values(image, labels, index=None):
...
-def mean_intensity(images, labels, index=None):
+    series_dict = {}
+    for name, data in data_dict.items():
+        by_roi, roi_list = _mean_intensity(data, labeled_array)
+        data = [by_roi[:, i] for i in range(by_roi.shape[1])]
+        series_dict[name] = pd.Series(data=data, index=['roi%s' % idx for idx in roi_list])
+    return pd.DataFrame(series_dict)
+    
+
+def _mean_intensity(images, labeled_array, index=None):
```
Comment (sameera2004) [inline skxray/core/roi.py]: @ericdill I saw you changed function name, you have to change that in test too, because I put a test for that function

**Comment by sameera2004** [comment]
Other than fixing the tests this function looks fine to me, when you fix it I will check it again

File: .travis.yml [hunk 2]
```diff
@@ -15,7 +15,8 @@ before_install:
 install:
   - export GIT_FULL_HASH=`git rev-parse HEAD`
   - conda update conda --yes
-  - conda create -n testenv --yes pip nose python=$TRAVIS_PYTHON_VERSION lmfit xraylib numpy scipy scikit-image netcdf4 six coverage
+  - conda create -n testenv --yes pip nose python=$TRAVIS_PYTHON_VERSION lmfit xraylib numpy scipy scikit-image netcdf4 six coverage pandas
+>>>>>>> MNT: Add pandas as a dep
```
Comment (tacaswell) [inline .travis.yml]: merge conflict remains.

**Comment by ericdill** [comment]
@sameera2004 can you review my changes?

**Comment by ericdill** [comment]
ignore failing travis for now. I want to get feedback from chx about this API before we sort out the best way to test these functions

File: .gitignore [hunk 3]
```diff
@@ -87,3 +87,6 @@ doc/resource/api/generated/
 *.txt
 *.zip
 *.jpg
+
+# ctags
+.tags*
```
Comment (sameera2004) [inline .gitignore]: You have added .gitignore

Comment (ericdill) [inline .gitignore]: What is your question here?

**Comment by ericdill** [comment]
superseded by #306

**Pull request closed without merging**